### PR TITLE
Feature/t72037 carrus keyboard

### PR DESCRIPTION
**Pull request by gagrison**
Task: https://phabricator.nsoft.ba/T72037
Connected with 7Shop task: https://phabricator.nsoft.ba/T71954

### CODE REVIEW

File: src/components/VirtualKeyboard.vue [hunk 1]
```diff
@@ -123,20 +125,28 @@ export default {
...
       this.overlay = dataset.overlay === 'true';
       this.headerText = dataset.headerText || '';
       this.predefinedValues = dataset.predefinedValues
         ? dataset.predefinedValues.split(',')
         : null;
       this.maxValue = dataset.maxValue || null;
+      this.defaultKeySet = dataset.defaultKeySet || 'default';
+
+      if (dataset.layout) {
+        if (!this.layout) {
```
Comment (aldinezi) [inline src/components/VirtualKeyboard.vue]: Will this work if you have multiple fields in one page, passing various layouts (eg. numeric, qwerty) ?

File: src/style/gravity/_keyboard.palette.scss [hunk 2]
```diff
@@ -1,32 +1,15 @@
...
-          .key {
-            background-color: $controlKeysBackground;
-
-            &:active {
-              color: $theme-normal-text;
-              background-color: lighten($theme-normal-background, 10);
-            }
-          }
-        }
-      }
```
Comment (stvedo) [inline src/style/gravity/_keyboard.palette.scss]: Is there some reason why is this entire style removed ?

**Comment by aldinezi** [comment]
This is not good in general. In the preview build, I see that the overall design is changed, which would break component desing that is already in production on terminal. :man_shrugging:

File: src/components/VirtualKeyboard.vue (hunk 1 above)
Comment (gagrison) [reply src/components/VirtualKeyboard.vue]: Will check it.

File: src/style/gravity/_keyboard.palette.scss (hunk 2 above)
Comment (gagrison) [reply src/style/gravity/_keyboard.palette.scss]: Then I guess we need design for numeric keyboard as well

Comment (aldinezi) [reply src/style/gravity/_keyboard.palette.scss]: Design should remain the same, or toggled between various clients. This current implementation is already in production on terminal. 
@dika-pro please define task so the new desing changes don't break the current implementation that is used in production on terminal. Also, if the new implementation requires changes in already deployed apps, please document it.

**Comment by dika-pro** [comment]
- This design is defined by product development and as they said it should be like this on all our platforms/channels
   - we will fix numeric keyboard bugs and adopt new style-guide
- I'll create separate task for update on terminal and SM Terminal side 
  - if there is a better approach let me know
- We (team seven) will talk tomorrow to product development about potential issues with BB before those changes even hit staging area.

**Comment by aldinezi** [comment]
> * This design is defined by product development and as they said it should be like this on all our platforms/channels
>   
>   * we will fix numeric keyboard bugs and adopt new style-guide
> * I'll create separate task for update on terminal and SM Terminal side
>   
>   * if there is a better approach let me know
> * We (team seven) will talk tomorrow to product development about potential issues with BB before those changes even hit staging area.

OK, we can stick with this new approach on layouts and layout names. We are looking forward to new fix commit for numpad desing. 
And then we'll do our part on adapting sm terminal implementation.

File: src/components/VirtualKeyboard.vue (hunk 1 above)
Comment (gagrison) [reply src/components/VirtualKeyboard.vue]: Fixed. Good catch.

**Pull request closed without merging**